src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs

```diff
@@ -29,7 +30,23 @@ public static AzureAuthenticationInfo GetConfiguredAzureAuthentication(IConfigur
                 authenticationConfiguration = new AuthenticationConfiguration();
             }
 
-            var applicationKey = configuration.GetValue<string>(EnvironmentVariables.Authentication.ApplicationKey);
+            string applicationKey;
+
+            if (!string.IsNullOrEmpty(authenticationConfiguration.SecretFilePath) && !string.IsNullOrEmpty(authenticationConfiguration.SecretFileName))
+            {
+                var filePath = Path.Combine(authenticationConfiguration.SecretFilePath, authenticationConfiguration.SecretFileName);
+
+                if (!File.Exists(filePath))
+                {
+                    throw new AuthenticationException("Invalid secret file path was configured for service principle authentication");
```

Comment:
```suggestion
                    throw new AuthenticationException("Invalid secret file path was configured for service principle authentication because it does not exist");
```